pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -2355,6 +2355,48 @@ public class Foo {        // perfect, both methods provided
       // return some hash value
     }
 }
+]]>
+        </example>
+    </rule>
+    <rule name="MissingEqualsOnComparable"
+          language="java"
+          since="7.16.0"
+          message="Classes implementing Comparable should override both equals() and hashCode()"
+          class="net.sourceforge.pmd.lang.java.rule.errorprone.MissingEqualsOnComparableRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_errorprone.html#missingequalsoncomparable">
+        <description>
+            Classes that implement Comparable must override both equals() and hashCode() to maintain consistency
+            between comparison and equality. The natural ordering should be consistent with equals.
+        </description>
+        <priority>3</priority>
+        <example>
+            <![CDATA[
+public class Bar implements Comparable<Bar> {  // poor - missing equals() and hashCode()
+    public int compareTo(Bar other) {
+        // some comparison
+    }
+}
+
+public class Baz implements Comparable<Baz> {  // poor - missing hashCode()
+    public int compareTo(Baz other) {
+        // some comparison
+    }
+    public boolean equals(Object o) {
+        // some comparison
+    }
+}
+
+public class Foo implements Comparable<Foo> {  // correct
+    public int compareTo(Foo other) {
+        // some comparison
+    }
+    public boolean equals(Object o) {
+        // some comparison
+    }
+    public int hashCode() {
+        // return hash code
```

Comment:
how to impl. this in rewrite? Its as simple just adding the methods, as its complex, as there is no generic impl. for these kind, the IDE needs to determine which fields to include for an generated impl., as this is individual. 

@timtebeek